### ISSUE TITLE
Warning when manually invoking __magic__ methods

### ISSUE THREAD

**Issue by MiguelGuthridge** [PyCQA/pylint#5936]
### Current problem

It is almost always considered a bad practice to manually invoke a `__magic__` method on a class.
```py
# Bad
my_str = my_object.__str__()
# Good
my_str = str(my_object)
```
A warning for this type of issue could be nice and instructive for people who are new to Python.

### Desired solution

A warning is given when Pylint is run on code that invokes `__magic__` methods on a class rather than using the builtin functions for it.

**Comment by jpy-git** [comment]
I'd like to give this one a go please 😄

**Comment by jpy-git** [comment]
I've got a basic implementation working but had a few questions for refinement:
- I'm only flagging magic methods not dunder attributes since they usually have genuine use cases. e.g. `my_instance_name = x.__class__.__name__` or `my_pkg_version = pkg.__version__`. Happy with that?
- `super().__init__(self, ...)`, `Foo.__init__(self, ...)`, `super().__new__(cls, ...)`, or `Foo.__new__(cls, ...)` are all pretty common use cases where people manually call the magic method (`.__init__(...)` shows up 65 times in functional and `.__new__(...)` shows up 13 times). Would you agree that I should exclude these from this check?
- There's one [test](https://github.com/PyCQA/pylint/blob/2a33b6742ec5cc0bc718ef6cc691ed16684b8cab/tests/functional/r/regression_02/regression_4982.py#L9) where `cls.__subclasses__()` is used. Currently I'm flagging this, but there's not really a good alternative for this dunder method and I'd argue someone using this probably knows what they're doing, so another potential exclusion?
- `.__init_subclass__` ([here](https://github.com/PyCQA/pylint/blob/2a33b6742ec5cc0bc718ef6cc691ed16684b8cab/tests/functional/i/init_subclass_classmethod.py#L6-L8)) seems like another potential exclusion?

**Comment by MiguelGuthridge** [comment]
@jpy-git thanks so much for your work! I think your decisions are reasonable - if there are functions or attributes where there isn't a function you should call to get them, then it should give the warning, but otherwise, it shouldn't.

One thing that could be worth consideration is whether to use an inclusion list or an exclusion list. I feel like using a whitelist might be better in terms of reducing false positives if there are new `__magic__` methods added. It might be also useful to provide the function that should be used instead within the error message, so that people know what to use instead.

For example (I'm not familiar enough with the standard error grammar used by Pylint, so reinterpret this as necessary):
```
>>> my_object.__str__()
Manually invoked magic method my_object.__str__(). Use str(my_object) instead
>>> a.__add__(b)
Manually invoked magic method a.__add__(b). Use a + b instead
```
